### PR TITLE
Added Lots of pages to osu!ppy

### PR DESCRIPTION
**Pull request by DarkVillager**
- Added the whole wiki
- Added seeing what you search (at  home page & search page)
- Added kudosu ranking
Search:
image
image
image
Some wiki pages:
image
image
image

All pages added (that I tested with, all other wiki pages also work):
https://osu.ppy.sh/home/search
https://osu.ppy.sh/p/kudosu
https://osu.ppy.sh/home/support
https://osu.ppy.sh/wiki/en/Installation
https://osu.ppy.sh/wiki/en/Registration
https://osu.ppy.sh/wiki/en/Chat_Console
https://osu.ppy.sh/wiki/en/BanchoBot
https://osu.ppy.sh/wiki/en/Guides
https://osu.ppy.sh/wiki/en/Glossary
https://osu.ppy.sh/wiki/en/Interface
https://osu.ppy.sh/wiki/en/Options
https://osu.ppy.sh/wiki/en/Visual_Settings
https://osu.ppy.sh/wiki/en/Shortcut_key_reference
https://osu.ppy.sh/wiki/en/osu!_Program_Files
https://osu.ppy.sh/wiki/en/osu!_File_Formats
https://osu.ppy.sh/wiki/en/Twitter
https://osu.ppy.sh/wiki/en/Beatmap_Editor
https://osu.ppy.sh/wiki/en/Submission
https://osu.ppy.sh/wiki/en/Modding
https://osu.ppy.sh/wiki/en/Beatmap_ranking_procedure
https://osu.ppy.sh/wiki/en/Mappers_Guild
https://osu.ppy.sh/wiki/en/Project_Loved
https://osu.ppy.sh/wiki/en/Ranking_Criteria
https://osu.ppy.sh/wiki/en/Contributor_Code_of_Conduct
https://osu.ppy.sh/legal/en
https://osu.ppy.sh/wiki/en/Performance_Troubleshooting
https://osu.ppy.sh/wiki/en/Silence
https://osu.ppy.sh/wiki/en/How_you_can_help!
https://osu.ppy.sh/wiki/en/Reporting_Bad_Behaviour
https://osu.ppy.sh/wiki/en/Beatmap_Spotlights
https://osu.ppy.sh/wiki/en/Contests
https://osu.ppy.sh/wiki/en/Tournaments
https://osu.ppy.sh/wiki/en/Contests
https://osu.ppy.sh/wiki/en/Community_Mentorship_Program
https://osu.ppy.sh/wiki/en/Projects
https://osu.ppy.sh/wiki/en/Organisations/osu!_UCI
https://osu.ppy.sh/wiki/en/osu!api
https://osu.ppy.sh/wiki/en/Bot_account
https://osu.ppy.sh/wiki/en/Brand_identity_guidelines
https://osu.ppy.sh/wiki/en/History_of_osu!
https://osu.ppy.sh/wiki/en/Mapping_and_Modding_Timeline
https://osu.ppy.sh/wiki/en/Sitemap
https://osu.ppy.sh/wiki/en/osu!_wiki/Contribution_guide
https://osu.ppy.sh/wiki/en/Article_styling_criteria
https://osu.ppy.sh/wiki/en/News_Styling_Criteria

### CODE REVIEW

File: websites/O/osu!ppy/presence.ts [hunk 1]
```diff
@@ -9,19 +9,30 @@ let beatmapTitle: string,
...
-    presenceData.details = "Viewing...";
-    presenceData.state = "The Homepage";
-    presenceData.smallImageKey = "searching";
+    inputSelected = document.querySelector(
+      "body > div:nth-child(23) > div > div > div.quick-search-input > div > input"
+    );
+    if (inputSelected) {
+      presenceData.details = "Searching for:";
+      presenceData.state =
+        inputSelected.value != "" ? inputSelected.value : "Nothing";
```
Comment (Slowlife01) [inline websites/O/osu!ppy/presence.ts:29]:
```suggestion
        inputSelected.value !== "" ? inputSelected.value : "Nothing";
```

File: websites/O/osu!ppy/presence.ts [hunk 2]
```diff
@@ -139,33 +156,36 @@ presence.on("UpdateData", async () => {
     presenceData.state = "Tournaments";
     presenceData.smallImageKey = "searching";
   } else if (document.location.pathname.startsWith("/home/search")) {
-    presenceData.details = "Searching...";
-    presenceData.smallImageKey = "searching";
+    inputSelected = document.querySelector("#search-input");
+    if (inputSelected) {
+      presenceData.details = "Searching for:";
+      presenceData.state =
+        inputSelected.value != "" ? inputSelected.value : "Nothing";
```
Comment (Slowlife01) [inline websites/O/osu!ppy/presence.ts:163]:
```suggestion
        inputSelected.value !== "" ? inputSelected.value : "Nothing";
```

File: websites/O/osu!ppy/presence.ts [hunk 3]
```diff
@@ -139,33 +156,36 @@ presence.on("UpdateData", async () => {
...
+      presenceData.smallImageKey = "searching";
+    } else {
+      presenceData.details = "Searching...";
+      presenceData.state = presenceData.smallImageKey = "searching";
```
Comment (Slowlife01) [inline websites/O/osu!ppy/presence.ts:167]:
```suggestion
      presenceData.state = "searching";
      presenceData.smallImageKey = "searching";
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
update your fork

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**